test/extensions/filters/network/mongo_proxy/proxy_test.cc

```diff
@@ -69,14 +72,19 @@ class MongoProxyFilterTest : public testing::Test {
     ON_CALL(runtime_.snapshot_, featureEnabled("mongo.logging_enabled", 100))
         .WillByDefault(Return(true));
 
+    EXPECT_CALL(read_filter_callbacks_, connection())
+        .WillRepeatedly(ReturnRef(read_filter_callbacks_.connection_));
+    EXPECT_CALL(read_filter_callbacks_.connection_, streamInfo())
+        .WillRepeatedly(ReturnRef(stream_info_));
+
     EXPECT_CALL(log_manager_, createAccessLog(_)).WillOnce(Return(file_));
     access_log_.reset(new AccessLog("test", log_manager_, dispatcher_.timeSystem()));
   }
 
   void initializeFilter() {
-    filter_ =
-        std::make_unique<TestProxyFilter>("test.", store_, runtime_, access_log_, fault_config_,
-                                          drain_decision_, generator_, dispatcher_.timeSystem());
+    filter_ = std::make_unique<TestProxyFilter>("test.", store_, runtime_, access_log_,
+                                                fault_config_, drain_decision_, generator_,
+                                                dispatcher_.timeSystem(), true);
```

Comment:
Do you have a test that makes sure dynamic data is not emitted when the config option is off?

Reply:
Added in 8423c17.